pyModeS/decoder/adsb.py

```diff
@@ -264,77 +232,31 @@ def nic_v2(msg, nic_a, nic_b, nic_c):
         nic_a (int or string): NIC supplement
         nic_b (int or srting): NIC supplement
         nic_c (int or string): NIC supplement
+
     Returns:
-        int: NIC number (from 0 to 11), -1 if not applicable
+        int or string: Horizontal Radius of Containment 
     """
     if typecode(msg) < 5 or typecode(msg) > 22:
         raise RuntimeError("%s: Not a surface position message (5<TC<8) \
                            airborne position message (8<TC<19), \
                            or airborne position with GNSS height (20<TC<22)" % msg)
 
-    tc = typecode(msg)
+    tc = typecode(msg)               
 
-    if nic_a in ['0', '1']:
-        nic_a = int(nic_a)
+    nic_df = pd.read_csv('/home/josmilrom/Libraries/pyModeS/pyModeS/decoder/adsb_ua_parameters/NIC_v2.csv', sep=',')
 
-    if nic_b in ['0', '1']:
-        nic_b = int(nic_b)
+    if tc in range(5,9):
+        nic_df_extract = [(nic_df.TC == tc) & (nic_df.NICa == nic_a) & (nic_df.NICc == nic_c)]
```

Comment:
must be: nic_df_extract =nic_df[(nic_df.TC == tc) & (nic_df.NICa == nic_a) & (nic_df.NICc == nic_c)]